### ISSUE TITLE
Allow user-defined metadata section in rtdc file

### ISSUE THREAD

**Issue by PinkShnack** [ZELLMECHANIK-DRESDEN/dclab#125]
It would be nice for a user to add their own metadata to the `rtdc_ds.config` attribute that can then be saved and reloaded later.

Currently, the config attribute is limited to the metadata defined in the `dclab.dfn` file. One can currently define a new `config` section but this section is not reloaded after saving the file (or it isn't saved at all?). I could not add key value pairs to this new section, however. Another option is to just add a h5py `attrs` to the file, but then this metadata is also not reloaded by dclab.

@paulmueller had a clean solution of just adding an empty "user" section to the `config` dictionary. This would automatically be useable for saving and reloading as it will be defined in `clab.dfn`. Then a user can just set their extra metadata by either:
```python
my_metadata = {
    "inlet": True,
    "channel": False,
    ...
}
ds.config["user"] = my_metadata
```
Or perhaps using `ds.config.update` is the more correct way:
```python
my_metadata = {
    "inlet": True,
    "channel": False,
    ...
}
ds.config.update({"user": my_metadata})
```

**Comment by paulmueller** [comment]
Both `__setitem__` and `update` should be implemented. While I think that the current implementation of `update` already does things recursively (nothing is overridden). There will probably also be some code-cleanup involved, because I didn't touch that code for a long time.

**Comment by paulmueller** [comment]
BTW: If this is only about "inlet" and "channel", you could also use `[setup]: "chip region"` (we only have to add "inlet").

**Comment by PinkShnack** [comment]
> BTW: If this is only about "inlet" and "channel", you could also use `[setup]: "chip region"` (we only have to add "inlet" as a valid option).

It isn't limited to these entries.

**Comment by PinkShnack** [comment]
@felix-r this is now implemented via #129 in the latest version of dclab.
